apps/user-ms/src/main/java/it/pagopa/selfcare/user/util/QueryUtils.java

```diff
@@ -0,0 +1,121 @@
+package it.pagopa.selfcare.user.util;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Sorts;
+import com.mongodb.client.model.Updates;
+import it.pagopa.selfcare.user.constant.SortEnum;
+import it.pagopa.selfcare.user.entity.UserInstitution;
+import jakarta.enterprise.context.ApplicationScoped;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentReader;
+import org.bson.Document;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.DocumentCodec;
+import org.bson.conversions.Bson;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor(access = AccessLevel.NONE)
+@ApplicationScoped
+public class QueryUtils {
+
+    private static final String CURRENT_PRODUCT = "current.";
+
+    public Document buildUpdateDocument(Map<String, Object> parameters) {
+        if (!parameters.isEmpty()) {
+            return bsonToDocument(Updates.combine(constructBsonUpdate(parameters)));
+        } else {
+            return new Document();
+        }
+    }
+
+    public Document buildQueryDocument(Map<String, Object> parameters) {
+        if (!parameters.isEmpty()) {
+            return bsonToDocument(Filters.and(constructBsonFilter(parameters)));
+        } else {
+            return new Document();
+        }
+    }
+
+    public Document buildSortDocument(String field, SortEnum order) {
+        if (SortEnum.ASC == order) {
+            return bsonToDocument(Sorts.ascending(field));
+        } else {
+            return bsonToDocument(Sorts.descending(field));
+        }
+    }
+
+    public Bson constructElementMatch(Map<String, Object> parameters) {
+        String arrayToMatch = Arrays.toString(parameters.keySet().stream().map(s -> s.split("\\."))
+                .toList()
+                .get(0));
+        return Filters.elemMatch(arrayToMatch, Filters.and(constructBsonFilter(parameters)));
+    }
+
+    /**
+     * The constructBsonFilter function takes a Map of parameters and returns a List of Bson objects.
+     * The function iterates over the entries in the parameter map, and for each entry it creates
+     * either an equality filter or a range filter depending on whether the key is &quot;from&quot; or &quot;to&quot;.
+     */
+    private List<Bson> constructBsonFilter(Map<String, Object> parameters) {
```

Comment:
Non si potrebbe fare in modo di costruire una stringa query evitando la costruzione del document e passandola direttamente al find?

`   public static String buildQuery(Map<String, Object> parameters) {
        return parameters.entrySet().stream()
                .map( entry -> {
                    if(entry.getValue() instanceof ArrayList<?>){
                        return entry.getKey() + "in :" + entry.getKey();
                    }
                    return entry.getKey() + "=:" + entry.getKey();
                })
                .collect( Collectors.joining(" and ") );
    }`